upf/simplifier.py

```diff
@@ -186,8 +196,115 @@ def walk_lt(self, expression: FNode, args: List[FNode]) -> FNode:
 
     def walk_fluent_exp(self, expression: FNode, args: List[FNode]) -> FNode:
         return self.manager.FluentExp(expression.fluent(), tuple(args))
+        
+    def walk_plus(self, expression: FNode, args: List[FNode]) -> FNode:
+        new_args_plus: List[FNode] = list()
+        accumulator = 0
+        #divide constant FNode and accumulate their value into accumulator
+        for a in args:
+            if a.is_int_constant() or a.is_real_constant():
+                accumulator = accumulator + a.constant_value()
+            elif a.is_plus():
+                for s in a.args():
+                    if s.is_int_constant() or s.is_real_constant():
+                        accumulator = accumulator + s.constant_value()
+                    else:
+                        new_args_plus.append(s)
+            else:
+                new_args_plus.append(a)
+        #if accumulator != 0 create it as a constant FNode and then add all the non-constant FNodes found
+        #else return 0 or all the non-constant FNodes found
+        if accumulator != 0:
+            if isinstance(accumulator, int):
+                fnode_constant_values = self.manager.Int(accumulator)
+            else:
+                fnode_constant_values = self.manager.Real(Fraction(accumulator))
+            fnode_acc = fnode_constant_values
+            for a in new_args_plus:
+                fnode_acc = self.manager.Plus(a, fnode_acc)
+            return fnode_acc
+        else: 
+            if len(new_args_plus) == 0:
+                return self.manager.Int(0)
```

Comment:
Maybe here we can return `self.manager.Real(Fraction(0))` when at least one of the constant args was a real.
@mikand what do you think?

Reply:
Given the current typing system, that auto-promotes integers to reals, it should be safe to return an integer, no?

Reply:
Yes, with the current typing system it is correct.
I was wondering if the proposed suggestion would be more robust to possible future changes in the typing system.

Reply:
@mikand so do you think it is better to keep `Int(0)`?

Reply:
For the time being, I would keep `Int()`. If the type system changes we will adapt.